config/ompi_fortran_check_iso_fortran_env_real16.m4

```diff
@@ -28,6 +28,22 @@ end program]])],
              [AS_VAR_SET(iso_fortran_env_var_real16, no)])
         AC_LANG_POP([Fortran])
        ])
+    AS_VAR_IF(iso_fortran_env_var_real16, [yes],
+          [AC_MSG_CHECKING([if Fortran compiler supports complex(ISO_FORTRAN_ENV:REAL16)])
+           AC_LANG_PUSH([Fortran])
+           AC_COMPILE_IFELSE([AC_LANG_SOURCE([[program check_for_iso_fortran_env_real16
+   use, intrinsic :: iso_fortran_env
+   complex(REAL16) :: var
+   var = 12.34
+end program]])],
+             [AC_MSG_RESULT([yes])],
+             [AC_MSG_RESULT([no])
+              AC_MSG_WARN([*** Fortran compiler supports real(REAL16) but])
+              AC_MSG_WARN([*** does *not* support complex(REAL16)])
+              AC_MSG_WARN([*** Disabling support for ISO_FORTRAN_ENV:REAL16])
```

Comment:
Should we just create another type / plumb through complex(real16) such that we *can* support real16 and *not* support complex(real16)?

Reply:
We can just turn OMPI_HAVE_FORTRAN_COMPLEX32 off and the datatype and op code should deal with having REAL16 but not COMPLEX32.

Reply:
But I'm fine with the solution proposed here, either you have both or none.

Reply:
I made a quick grep and unless I missed something, this is only used in `MPI_Sizeof()`.
So though it would be ideal to handle compilers (e.g. `NVHPC`) that support `real(REAL16)` but not `complex(REAL16)`, I do not think this is worth the hassle and fixing compilation by disabling `ISO_FORTRAN_ENV:REAL16` altogether is good enough for the time being.
Of course anyone is free to improve this PR.

Reply:
It is also used in the datatype and in the MPI Op via OMPI_HAVE_FORTRAN_COMPLEX32. But I would be surprised if it had any users. For the outcome, I'm with you this PR is good enough.